pkg/client/fake/client_test.go

```diff
@@ -1590,6 +1591,28 @@ var _ = Describe("Fake client", func() {
 		Expect(objOriginal.Status.Phase).ToNot(Equal(actual.Status.Phase))
 	})
 
+	It("should be able to change typed objects that have a scale subresource on patch", func() {
```

Comment:
I have to say that I am amazed this works at all. It ends up patching the original object, not some subresource which happens to work if the original object has a `.spec.replicas` field, which doesn't have to be the case. I'll accept this patch, but I will also note that the `scale` subresource isn't really supported in the fake client and just coincidentally works under some specific coincidences

Reply:
> It ends up patching the original object, not some subresource which happens to work if the original object has a .spec.replicas field, which doesn't have to be the case. I'll accept this patch, but I will also note that the scale subresource isn't really supported in the fake client and just coincidentally works under some specific coincidences

This is true for any subresource other than status.